File: src/java/io/BufferedReader.ts
import { Stream } from '../util/stream/Stream.js'
import { Reader } from './Reader.js'

export class BufferedReader extends Reader {
  // Constructors
  public constructor(...args: [Reader] | [Reader, number]) {
    super()
    // reader: Reader
    if (args.length == 1) {
      return new Packages.java.io.BufferedReader(args[0])
    } else {
      // reader: Reader, int: number
      return new Packages.java.io.BufferedReader(args[0], args[1])
    }
  }
  // Methods
  private ensureOpen(): void {
    this.ensureOpen()
  }
  private fill(): void {
    this.fill()
  }
  private read1(c: String[], int: number, int_1: number): number {
    return this.read1(c, int, int_1)
  }
  public close(): void {
    this.close()
  }
  public lines(): Stream {
    return this.lines()
  }
  public mark(int: number): void {
    this.mark(int)
  }
  public markSupported(): boolean {
    return this.markSupported()
  }
  public read(...args: [] | [String[], number, number]): number {
    if (args.length == 0) {
      return this.read()
    } else {
      return this.read(args[0], args[1], args[2])
    }
  }
  public readLine(...args: [] | [boolean]): string {
    if (args.length == 0) {
      return this.readLine()
    } else {
      return this.readLine(args[0])
    }
  }
  public ready(): boolean {
    return this.ready()
  }
  public reset(): void {
    this.reset()
  }
  public skip(lon: number): number {
    return this.skip(lon)
  }
  // Fields
  private cb: String[]
  private in: Reader
  private markedChar: number
  private markedSkipLF: boolean
  private nChars: number
  private nextChar: number
  private readAheadLimit: number
  private skipLF: boolean
  private static INVALIDATED: number
  private static UNMARKED: number
  private static defaultCharBufferSize: number
  private static defaultExpectedLineLength: number
}

File: src/java/io/BufferedReader.java
public class java.io.BufferedReader extends java.io.Reader
{
    // Constructors
    public java.io.BufferedReader(java.io.Reader);
    public java.io.BufferedReader(java.io.Reader, int);
    // Methods
    private void ensureOpen();
    private void fill();
    private int read1([C, int, int);
    public void close();
    public java.util.stream.Stream lines();
    public void mark(int);
    public boolean markSupported();
    public int read();
    public int read([C, int, int);
    public java.lang.String readLine();
    java.lang.String readLine(boolean);
    public boolean ready();
    public void reset();
    public long skip(long);
    // Fields
    private [C cb;
    private java.io.Reader in;
    private int markedChar;
    private boolean markedSkipLF;
    private int nChars;
    private int nextChar;
    private int readAheadLimit;
    private boolean skipLF;
    private static final int INVALIDATED;
    private static final int UNMARKED;
    private static int defaultCharBufferSize;
    private static int defaultExpectedLineLength;
}